test_conformance/c11_atomics/common.h

```diff
@@ -807,16 +803,19 @@ std::string CBasicTest<HostAtomicType, HostDataType>::KernelCode(cl_uint maxNumD
       "    {\n";
     if(aTypeName == "atomic_flag")
     {
-      code +=
-        "      if(finalDest[dstItemIdx])\n"
-        "        atomic_flag_test_and_set(destMemory+dstItemIdx);\n"
-        "      else\n"
-        "        atomic_flag_clear(destMemory+dstItemIdx);\n";
+        code +=
+            "      if(finalDest[dstItemIdx])\n"
+            "        atomic_flag_test_and_set_explicit(destMemory+dstItemIdx, "
+            "memory_order_relaxed, memory_scope_work_group);\n"
+            "      else\n"
+            "        atomic_flag_clear_explicit(destMemory+dstItemIdx, "
+            "memory_order_relaxed, memory_scope_work_group);\n";
     }
     else
     {
-      code +=
-        "      atomic_store(destMemory+dstItemIdx, finalDest[dstItemIdx]);\n";
+        code += "      atomic_store_explicit(destMemory+dstItemIdx, "
+                "finalDest[dstItemIdx], memory_order_relaxed, "
+                "memory_scope_work_group);\n";
     }
     code +=
       "    }\n"
```

Comment:
Should the barrier on the next line be `CLK_LOCAL_MEM_FENCE` for the `LocalMemory()` case but be `CLK_GLOBAL_MEM_FENCE` otherwise?

Or could just `barrier(CLK_LOCAL_MEM_FENCE | CLK_GLOBAL_MEM_FENCE);` if that's simpler.

Reply:
I'm not entirely sure what the other cases are testing, do you mean [here](https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_conformance/c11_atomics/common.h#L841), [here](https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_conformance/c11_atomics/common.h#L854) and [here](https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_conformance/c11_atomics/common.h#L872).

Reply:
I was actually referring to the one [here](https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_conformance/c11_atomics/common.h#L823).

But my comment no longer makes sense to me.  I think I got mixed up with the nesting, I'll just resolve this point.